# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/fontconfig/hotfix-npe.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---


Null Pointer dereference on at least sparc6432

--- fontconfig-2.17.1/src/fcmatch.c.vanilla	2025-08-14 11:19:36.885000000 +0200
+++ fontconfig-2.17.1/src/fcmatch.c	2025-08-14 11:19:48.193267554 +0200
@@ -1190,7 +1190,7 @@
 	     * This way may not work if someone has own list of application fonts
 	     * That said, just to reduce the cost for lookup so far.
              */
-	    if (config->prefer_app_fonts && s != config->fonts[FcSetApplication]) {
+	    if (config && config->prefer_app_fonts && s != config->fonts[FcSetApplication]) {
 		newp->score[PRI_ORDER] += 1000;
 	    }
 	    if (FcDebug() & FC_DBG_MATCHV) {
